packages/generator-rest/src/generator.ts

```diff
@@ -13,71 +13,128 @@ import {
   removeSync,
   writeJsonSync
 } from 'fs-extra';
+import { Project } from 'ts-morph';
 import { GeneratorOptions } from './generator-cli';
+import { projectOptions, sourceFile } from './utils';
+import { toOpenApiModel } from './parse-open-api-json';
+import { OpenApiServiceMetadata } from './open-api-types';
+import { requestBuilderSourceFile } from './request-builder/file';
 
 const logger = createLogger({
   level: 'info',
+  package: 'generator',
   messageContext: 'rest-generator'
 });
 
-export async function generateRest(options: GeneratorOptions): Promise<any> {
+export async function generateRest(options: GeneratorOptions): Promise<void> {
   cleanDirectory(options.outputDir);
 
   const files = readdirSync(options.inputDir);
   const pathToTemplates = resolve(__dirname, '../templates');
   const pathToMustacheValues = join(__dirname, '../mustache-values.json');
 
-  return Promise.all(
-    files.map(async file => {
-      const folderForService = getFolderForService(options.outputDir, file);
-      mkdirSync(folderForService);
-
-      const adjustedOpenApiContent = getAdjustedOpenApiFile(
-        join(options.inputDir, file),
-        getServiceNameCamelCase(file)
-      );
-      const pathToAdjustedOpenApiDefFile = join(
-        folderForService,
-        'open-api.json'
-      );
-      writeJsonSync(pathToAdjustedOpenApiDefFile, adjustedOpenApiContent);
-
-      const generationArguments = [
-        'openapi-generator',
-        'generate',
-        '-i',
-        pathToAdjustedOpenApiDefFile,
-        '-g',
-        'typescript-axios',
-        '-o',
-        folderForService,
-        '-t',
-        pathToTemplates,
-        '--api-package',
-        'api',
-        '--model-package',
-        'model',
-        '--config',
-        pathToMustacheValues,
-        '--skip-validate-spec'
-      ];
+  const project = new Project(projectOptions());
+
+  const openApiServiceMetadata = await Promise.all(
+    files.map(async file =>
+      generateOneApi(file, options, pathToTemplates, pathToMustacheValues)
+    )
+  );
+  openApiServiceMetadata.map(metadata =>
+    generateSourcesForService(metadata, project, options)
+  );
+  await project.save();
+}
 
-      logger.info(`Argument for openapi generator ${generationArguments}`);
-
-      try {
-        const response = await execa('npx', generationArguments).catch(err => {
-          logger.error('In exception block 2');
-          logger.error(err);
-        });
-        if (response !== undefined) {
-          logger.info(`Generated the client ${response.stdout}`);
-        }
-      } catch (err) {
-        logger.error('In exception block');
-        logger.error(err);
-      }
-    })
+async function generateOneApi(
+  file: string,
+  options: GeneratorOptions,
+  pathToTemplates: string,
+  pathToMustacheValues: string
+) {
+  const folderForService = getFolderForService(options.outputDir, file);
+  mkdirSync(folderForService);
```

Comment:
Let's refer to directories rather then folders. Folder can be ambiguous (which it is not here, but anyways).